File: server/src/game/tools/conf/output/go/LevelConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type LevelConfigElem struct {
	Id int
	Exp int
	
}

type LevelConfig0 struct {
	Dic map[int]*LevelConfigElem
	Arr []*LevelConfigElem
}

var (
	LevelConfig LevelConfig0
)

func loadLevelConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	LevelConfig.Dic = make(map[int]*LevelConfigElem)
	LevelConfig.Arr = make([]*LevelConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &LevelConfig.Arr)
	for _,conf := range LevelConfig.Arr{
		LevelConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/game/conf/LevelConfig.ts
class LevelConfig {
    private static _dic:Dictionary<LevelConfig> = new Dictionary<LevelConfig>();
    private static _arr:Array<LevelConfig> = new Array<LevelConfig>();

	 public Id :number ;
     public Exp :number ;
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<LevelConfig>{
        return  LevelConfig._dic;
    }
    public static get Arr():Array<LevelConfig>{
        return LevelConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:LevelConfig = <LevelConfig>json[i];
				LevelConfig._arr.push(config);
				LevelConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("LevelConfig loader fail!");
        }
    }
}